### PR TITLE
🚋 ♻️ src/preact: start the typing train

### PR DESCRIPTION
**Pull request by samouri**
**summary**
- Adds a basic `tsconfig.json` to the `src/preact` directory. Initially only checking the `src/preact/index.js` file.
- Removes the now unnecessary extern file.

### CODE REVIEW

File: src/preact/index.js [hunk 1]
```diff
@@ -8,15 +8,19 @@
  */
 
 // Importing directly from preact/dom is a hack to allow us to mimic the structure of react.
+
+/* eslint-disable */
 // @ts-ignore
-import * as preactDOM from 'preact/dom'
+import * as preactDOM from 'preact/dom';
```
Comment (rcebulko) [inline src/preact/index.js:15]: Instead of this, do `import {hydrate, render};` then type the exports

File: src/preact/index.js [hunk 2]
```diff
@@ -1,147 +1,42 @@
...
+export const hydrate = preactDOM.hydrate;
 
-/**
- * @param {function():(function():undefined|undefined)} effect
- * @param {!Array<*>=} opt_deps
- */
-export function useEffect(effect, opt_deps) {
-  hooks.useEffect(effect, opt_deps);
-}
+/** @type {typeof import('preact').hydrate} */
```
Comment (jridgewell) [inline src/preact/index.js:19]:
```suggestion
/** @type {typeof import('preact').render} */
```

Comment (samouri) [reply src/preact/index.js]: whoops, copy/paste mistake

**Review by jridgewell — APPROVED** [review]

File: src/preact/index.js (hunk 1 above)
Comment (samouri) [reply src/preact/index.js:15]: What is the syntax for that? I don't think `export hydrate: TYPE` works

Comment (rcebulko) [reply src/preact/index.js:15]: Basically the same way you have it already:

```js
import {hydrate, render} from 'preact/dom';
/** @type {typeof import('preact').hydrate} */
export hydrate;
/** @type {typeof import('preact').render} */
export render;
```

Comment (samouri) [reply src/preact/index.js:15]: I don't believe that is valid syntax. Export requires either a declaration (`export const hydrate = ...`) or for it to be within a set of exports (`export {hydrate}`). 

I'll merge as-is, but if you figure out a way to make this work cleanly I'm all for it!

**Review by kristoferbaxter — APPROVED** [review]

File: src/preact/index.js (hunk 1 above)
Comment (rcebulko) [reply src/preact/index.js:15]: Could we add `preact-dom.d.ts` with

```typescript
declare module 'preact/dom' {
  export const hydrate: typeof import('preact').hydrate;
  export const render: typeof import('preact').render;
}
```

**Review by rcebulko — APPROVED** [review]

**Comment by samouri** [comment]
@rcebulko, I believe your idea in https://github.com/ampproject/amphtml/pull/37242#discussion_r778249525 works!

**Pull request merged**